kitsune/announcements/models.py

```diff
@@ -85,8 +82,8 @@ def _visible_query(cls, **query_kwargs):
 
 
 def connector(sender, instance, created, **kw):
-    # Only email new announcements in a group. We don't want to email everyone.
-    if created and instance.group and instance.send_email:
+    # Only email new announcements in specified groups. We don't want to email everyone.
+    if created and instance.groups and instance.send_email:
```

Comment:
Since `instance.groups` is always true -- I think it's an instance of `QuerySet` -- this should be `instance.groups.count()` instead. However, I've only just realized there's a fundamental problem now that we're using `groups`, a many-to-many relationship, rather than `group`. The `post_save` signal will only be fired when the `Announcement` is saved, but that's always **before** any groups have been added, so no emails will ever be sent.

It seems that we need to know when the Admin has finished adding all of the groups to the `Announcement`, and only then we can send the emails, but I'm not sure of the best way to do that. In any case, I don't think we can use the `post_save` signal with this `connector` any longer. I suppose we have to create a new signal that we fire once the admin has finished adding all of the groups?